### PR TITLE
[gdb] Handle anonymous fields in GetAllFields and LookupField

### PR DESCRIPTION
**Pull request by cbiesinger**
This comes up here, for example:
https://cs.chromium.org/chromium/src/third_party/blink/renderer/core/layout/layout_box.h?q=LayoutBox&sq=package:chromium&g=0&l=1812

Without this change we throw an exception, so we don't get any LayoutBox
fields. Longer-term, we need to find a way to handle anonymous unions.

### CODE REVIEW

File: server/JsDbg.Gdb/JsDbg.py [hunk 1]
```diff
@@ -151,7 +151,10 @@ def GetAllFields(module, type, includeBaseTypes):
             # Field is static
             continue
 
-        
+        if not field.name:
+            # Field is anonymous, we have no way to access it.
```
Comment (sanketj) [inline server/JsDbg.Gdb/JsDbg.py]: I'm curious...how do we get this information in the debugger (GDB)?

Comment (cbiesinger) [reply server/JsDbg.Gdb/JsDbg.py]: Looks like you access it as if it were a direct member of LayoutBox (like in C++):
(gdb) p $lo
$3 = (blink::LayoutBlockFlow *) 0x3587f6e22ab8
(gdb) p $lo->inline_box_wrapper_
$4 = (blink::InlineBox *) 0x0

So we could probably just consider them part of the parent, as long as we're careful with the bit offset...

Comment (cbiesinger) [reply server/JsDbg.Gdb/JsDbg.py]: OK, please hold off on merging this, I really need to resolve this -- blink::Length is pretty useless without supporting anonymous unions

Comment (sanketj) [reply server/JsDbg.Gdb/JsDbg.py]: Ok, will un-approve for now so it stays on my list.

Comment (cbiesinger) [reply server/JsDbg.Gdb/JsDbg.py]: Done -- please take another look!

**Pull request merged**